cmd2/parsing.py

```diff
@@ -13,6 +13,55 @@
 from . import utils
 
 
+@attr.s(frozen=True)
+class MacroArg:
+    """
+    Information used to replace or unescape arguments in a macro value when the macro is resolved
+    Normal argument syntax  : {5}
+    Escaped argument syntax: {{5}}
+    """
+    # The starting index of this argument in the macro value
+    start_index = attr.ib(validator=attr.validators.instance_of(int))
+
+    # The number string that appears between the braces
+    # This is a string instead of an int because we support unicode digits and must be able
+    # to reproduce this string later
+    number_str = attr.ib(validator=attr.validators.instance_of(str))
+
+    # Tells if this argument is escaped and therefore needs to be unescaped
+    is_escaped = attr.ib(validator=attr.validators.instance_of(bool))
+
+    # Pattern used to find normal argument
+    # Digits surrounded by exactly 1 brace on a side and 1 or more braces on the opposite side
+    # Match strings like: {5}, {{{{{4}, {2}}}}}
+    macro_normal_arg_pattern = re.compile(r'(?<!\{)\{\d+\}|\{\d+\}(?!\})')
```

Comment:
Does "{" or "}" need to be escaped in a RegEx?  I don't think so.  This applies to the regex a few lines below as well.